### PR TITLE
Reminder UI + Storage

### PR DESCRIPTION
**Pull request by TSAI-HSIAO-HAN**
- [x] Create Reminder UI
- [x] Create Reminder Edit, Delete Command(without Storage)
- [x] Refactor Reminder Structure(Change dependency)
- [x] Implement Storage
- Add testing for Storage(do in V1.4)

### CODE REVIEW

File: src/main/java/seedu/mark/storage/JsonAdaptedReminder.java [hunk 1]
```diff
@@ -11,20 +18,24 @@
 public class JsonAdaptedReminder {
 
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Reminder's %s field is missing!";
+    private static final String DATE_FORMATTER = "dd/MM/yyyy HHmm";
+    protected static final String MESSAGE_INVALID_TIME_FORMAT =
+            "Invalid time format! Please use the following format: " + DATE_FORMATTER;
```
Comment (dorcastan) [inline src/main/java/seedu/mark/storage/JsonAdaptedReminder.java]: Not very important, but easier for user to understand (when the whole json file is read and errors appear).
```suggestion
            "Reminder time has the wrong format! Please use the following format: " + DATE_FORMATTER;
```

File: src/main/java/seedu/mark/logic/commands/DeleteReminderCommand.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        List<Reminder> lastShownList = model.getReminders();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size() || targetIndex.getOneBased() < 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_REMINDER_DISPLAYED_INDEX);
+        }
+
+        Reminder reminderToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        model.removeReminder(reminderToDelete);
```
[the remark this answers is not in the document]
Comment (TSAI-HSIAO-HAN) [reply src/main/java/seedu/mark/logic/commands/DeleteReminderCommand.java]: We change the index in UG to the index of reminder in the dashboard panel. I think I will add the index number of reminder on the UI in v1.4.

File: src/main/java/seedu/mark/model/reminder/Reminder.java [hunk 3]
```diff
@@ -81,22 +81,20 @@ public boolean equals(Object other) {
...
-        return Objects.hash(bookmark, remindTime, note);
+        return Objects.hash(url, remindTime, note);
     }
 
     @Override
     public String toString() {
 
         final StringBuilder builder = new StringBuilder();
         builder.append(getNote())
-                .append(" Bookmark: ")
```
[the remark this answers is not in the document]
Comment (TSAI-HSIAO-HAN) [reply src/main/java/seedu/mark/model/reminder/Reminder.java]: Ok.

File: src/main/java/seedu/mark/logic/commands/EditReminderCommand.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+import seedu.mark.storage.Storage;
+
+/**
+ * Edits the details of an existing reminder in Mark.
+ */
+public class EditReminderCommand extends Command {
+
+    public static final String COMMAND_WORD = "reminder-edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the reminder identified."
```
Comment (Na-Nazhou) [inline src/main/java/seedu/mark/logic/commands/EditReminderCommand.java]:
```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the reminder identified."
```

File: src/test/java/seedu/mark/logic/commands/DeleteReminderCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.model.Model;
+import seedu.mark.model.ModelManager;
+import seedu.mark.model.UserPrefs;
+import seedu.mark.model.reminder.Reminder;
+import seedu.mark.storage.StorageStub;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) for
+ * {@code DeleteCommand}.
```
Comment (Na-Nazhou) [inline src/test/java/seedu/mark/logic/commands/DeleteReminderCommandTest.java]:
```suggestion
 * {@code DeleteReminderCommand}.
```

**Pull request merged**